pkg/odo/cli/component/describe.go

```diff
@@ -47,18 +47,24 @@ func (do *DescribeOptions) Complete(name string, cmd *cobra.Command, args []stri
 
 // Validate validates the describe parameters
 func (do *DescribeOptions) Validate() (err error) {
-	if do.Context.Project == "" || do.Application == "" {
+	if do.Application == "" {
 		return odoutil.ThrowContextError()
 	}
 
+	if (len(do.componentName) <= 0 || len(do.Application) <= 0) && !do.LocalConfigInfo.Exists() {
```

Comment:
We're running `odoutil.ThrowContextError()` couiple of lines above if `do.Application` is empty. I'm wondering how checking for `do.Application` here and in the `if` condition below helps.

Reply:
Also, couple of lines back, we're checking:
```go
if do.Application == ""
```
but here we're using:
```go
len(do.ComponentName) <=0
```
to figure if it's an empty string.

I don't think there's an idiomatic way of doing this but within the same function (`Validate`), can we have uniformity?

Reply:
sure

Reply:
@girishramnani In case you missed this question - https://github.com/openshift/odo/pull/4501#discussion_r591275942

Reply:
yeah you are right - we dont need that check